### PR TITLE
Try with resources cleanup

### PR DESCRIPTION
**Pull request by pfistererm**
Try with resources cleanup

## Type of request

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactoring 
- [ ] Documentation or documentation changes

## Concept 
try-with-resources  should be used whenever possible.
#### Initiator
Matthias Pfisterer

#### How has this Tested?
Compiling and running be executing licensescout.quickstart

## Checklist:

- [x] All the commits are signed.
- [x] My code follows the code style of this project.
- [x] I agree with die CLA.
- [x] I have read the CONTRIBUTING docs.
- [x] I have added/updated necessary documentation (if appropriate).

### CODE REVIEW

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/ArtifactServerUtil.java [hunk 1]
```diff
@@ -31,9 +31,10 @@
 import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
 
 /**
- * Utility methods related to accessing a server that is caching maven central.
+ * Utility methods related to accessing a server that is caching Maven central.
  * 
- * <p>The instance of this class checks if the server is reachable under the configured URL. 
+ * <p>The instance of this class checks if the server is reachable under the
+ * configured URL.</p>
```
Comment (d-gs) [inline org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/ArtifactServerUtil.java]: Please, could you revert all those changes that are just formatting, to make easier the review?

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/LicenseCheckedList.java [hunk 2]
```diff
@@ -386,299 +386,299 @@ private static Provider getProviderFromId(final Providers providers, final Strin
         return provider;
     }
 
-	private class ArchiveSaxHandler extends AbstractSaxHandler {
-		
-	    private static final String ELEMENT_ARCHIVES = "archives";
```
Comment (d-gs) [inline org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/LicenseCheckedList.java]: Also, could you please revert indentation changes?

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/LicenseStoreData.java [hunk 3]
```diff
@@ -190,77 +190,77 @@ private void addNoManualInformationLicense(final ILFLog log) {
     }
 
     /**
-	 * Reads license URL mappings from a CSV file.
```
Comment (d-gs) [inline org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/LicenseStoreData.java]: Same here.

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/JarUtil.java [hunk 4]
```diff
@@ -38,32 +38,33 @@ private JarUtil() {
     }
 
     /**
-     * Tries to obtains meta information for a MANIFEST.MF that is available as a file.
+     * Tries to obtains meta information for a MANIFEST.MF that is available as a
+     * file.
```
Comment (d-gs) [inline org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/JarUtil.java]: Same about javadoc wrapping.

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/LicenseCheckedList.java (hunk 2 above)
Comment (pfistererm) [reply org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/LicenseCheckedList.java]: We should agree an a unique formatting and enforce it by providing a eclipse formatter rules file. 
For changing the indentation, I actually used the A3k formatting rules that were used so far.

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/JarUtil.java (hunk 4 above)
Comment (pfistererm) [reply org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/JarUtil.java]: Reverted some, but kept very long lines broken up.

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/LicenseStoreData.java (hunk 3 above)
Comment (pfistererm) [reply org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/LicenseStoreData.java]: Reverted some, but kept very long lines broken up.

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/ArtifactServerUtil.java (hunk 1 above)
Comment (pfistererm) [reply org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/ArtifactServerUtil.java]: Reverted some, but kept very long lines broken up.

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/CryptUtil.java [hunk 5]
```diff
@@ -45,11 +45,8 @@ private CryptUtil() {
      * @throws IOException
      */
     public static byte[] calculateMessageDigest(final File file) throws IOException {
-        final FileInputStream fis = new FileInputStream(file);
-        try {
+        try (final FileInputStream fis = new FileInputStream(file);) {
```
Comment (davu-all) [inline org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/CryptUtil.java]: Please remove the unnecessary semicolon. thanks

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/JarUtil.java [hunk 6]
```diff
@@ -42,28 +42,28 @@ private JarUtil() {
...
      * @throws IOException
      * 
      * @see #getArchiveMetaInformationFromManifest(File, ILFLog)
      * @see #getArchiveMetaInformationFromManifest(InputStream, ILFLog)
      */
     public static ArchiveMetaInformation getArchiveMetaInformationFromManifestFile(final File file, final ILFLog log)
             throws IOException {
-        final InputStream is = new FileInputStream(file);
-        try {
+        try (final InputStream is = new FileInputStream(file);) {
```
Comment (davu-all) [inline org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/JarUtil.java]: Please remove the unnecessary semicolon. thanks

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/JarUtil.java [hunk 7]
```diff
@@ -72,29 +72,29 @@ public static ArchiveMetaInformation getArchiveMetaInformationFromManifestFile(f
...
      * 
      * @see #getArchiveMetaInformationFromManifestFile(File, ILFLog)
      * @see #getArchiveMetaInformationFromManifest(InputStream, ILFLog)
      */
     public static ArchiveMetaInformation getArchiveMetaInformationFromManifest(final File file, final ILFLog log)
             throws IOException {
-        final JarFile jarFile = new JarFile(file);
         final String sourceForErrorText = file.getAbsolutePath();
-        try {
+        try (final JarFile jarFile = new JarFile(file);) {
```
Comment (davu-all) [inline org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/JarUtil.java]: Please remove the unnecessary semicolon. thanks

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/JarUtil.java [hunk 8]
```diff
@@ -120,13 +123,10 @@ public static ArchiveMetaInformation getArchiveMetaInformationFromManifest(final
     public static ArchiveMetaInformation getArchiveMetaInformationFromManifest(final InputStream inputStream,
                                                                                final ILFLog log)
             throws IOException {
-        final JarInputStream jarFile = new JarInputStream(inputStream);
         final String sourceForErrorText = "input stream";
-        try {
+        try (final JarInputStream jarFile = new JarInputStream(inputStream);) {
```
Comment (davu-all) [inline org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/util/JarUtil.java]: Please remove the unnecessary semicolon. thanks

File: org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/ArtifactServerUtil.java [hunk 9]
```diff
@@ -75,8 +75,7 @@ private boolean checkAccess() {
             final URL url = new URL(urlString);
             final URLConnection urlConnection = url.openConnection();
             urlConnection.setConnectTimeout(connectTimeoutInMilliseconds);
-            InputStream is = urlConnection.getInputStream();
-            is.close();
+            try (InputStream is = urlConnection.getInputStream();) {}
```
Comment (davu-all) [inline org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/ArtifactServerUtil.java]: Please remove the unnecessary semicolon. thanks

**Pull request merged**